File: src/dataset/c/triangle_Exercism_HeitorMP/openAI/ChatGPT4_triangle_v2.c
#include "triangle.h"

int is_valid_triangle(triangle_t sides) {
    return (sides.a > 0 && sides.b > 0 && sides.c > 0 &&
            sides.a + sides.b >= sides.c &&
            sides.b + sides.c >= sides.a &&
            sides.a + sides.c >= sides.b);
}

int is_equilateral(triangle_t sides) {
    return (is_valid_triangle(sides) &&
            sides.a == sides.b && sides.b == sides.c);
}

int is_isosceles(triangle_t sides) {
    return (is_valid_triangle(sides) &&
            (sides.a == sides.b || sides.b == sides.c || sides.a == sides.c));
}

int is_scalene(triangle_t sides) {
    return (is_valid_triangle(sides) &&
            sides.a != sides.b && sides.b != sides.c && sides.a != sides.c);
}

File: src/dataset/typescript/triangle_Exercism-typescript-alemarr/openAI/ChatGPT4_triangle_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Triangle = void 0;
class Triangle {
    constructor(a, b, c) {
        this.a = a;
        this.b = b;
        this.c = c;
        this.sides = [a, b, c].sort((x, y) => x - y);
    }
    get isEquilateral() {
        return this.isValid() && this.a === this.b && this.b === this.c;
    }
    get isIsosceles() {
        return this.isValid() && (this.a === this.b || this.b === this.c || this.a === this.c);
    }
    get isScalene() {
        return this.isValid() && this.a !== this.b && this.b !== this.c && this.a !== this.c;
    }
    isValid() {
        return this.sides[0] > 0 && this.sides[0] + this.sides[1] > this.sides[2];
    }
}
exports.Triangle = Triangle;
